File: validParanthesis/C++/validParanthesis.cpp
#include<iostream>
#include<stack>
#include<string>

using namespace std;


bool isValidParanthesis(string s){
    stack<char> stk;
    //we are having only these characters "({[]})" in the string so i companring directly
    
    for(char c:s){
        if( c =='(' || c == '{' || c == '[' ){
            stk.push(c);
        }
        else{
            if(stk.empty()){
                cout << "doing the false";
                return false;
            }
            if( c ==']' && stk.top()=='[' || c =='}' && stk.top()=='{' || c ==')' && stk.top()=='(' ) stk.pop();

            else return false;
        }
    }
    cout << stk.empty();
    return stk.empty();
}



int main(){
    string str = "()[]";   
    isValidParanthesis(str);
    return 0;
}

File: validParanthesis/Js/validParanthesis.js
const statement = "()]";
console.log(isValidParanthesis(statement));

function isValidParanthesis(str) {
  let stk = [];
  for (let c of str) {
    if (c === "(" || c === "{" || c === "[") stk.push(c);
    else {
      if (stk.length === 0) return false;
      if (
        (c === ")" && stk[stk.length - 1] === "(") ||
        (c === "}" && stk[stk.length - 1] === "{") ||
        (c === "]" && stk[stk.length - 1] === "[")
      )
        stk.pop();
      else return false;
    }
  }
  return stk.length === 0;
}
